pareto/utilities/get_data.py

```diff
@@ -17,32 +17,112 @@
 Authors: PARETO Team (Andres J. Calderon, Markus G. Drouven)
 """
 
+import logging
+import warnings
+from pathlib import Path
+from typing import Any
+from typing import Dict
+from typing import Iterable
+from typing import List
+from typing import Union
+
 import pandas as pd
 import requests
 import numpy as np
 import warnings
 
 
-def _read_data(_fname, _set_list, _parameter_list):
+_logger = logging.getLogger(__name__)
+
+
+try:
+    _dataframe_map = pd.DataFrame.map
+except AttributeError:
+    # compatibility with pandas 2.0.x
+    _dataframe_map = pd.DataFrame.applymap
+
+
+class DataLoadingError(ValueError):
+    def __init__(
+        self,
+        errors: Dict[str, Exception],
+    ):
+        self.errors = dict(errors)
+        self.summary = "\n".join(self._get_summary_lines())
+        super().__init__(self.summary)
+
+    def _get_summary_lines(self) -> List[str]:
+        max_name_len = max(len(name) for name in self.errors)
+
+        lines = ["Data loading failed for the following sheets: "]
+        for name, err in self.errors.items():
+            lines.append(f"\t{name:{max_name_len}}\t{err}")
+        return lines
+
+
+def _sheets_to_dfs(
+    src: Union[str, Path],
+    sheets: Iterable[str],
+    raises: bool = True,
+    **kwargs,
+) -> Dict[str, pd.DataFrame]:
+    out = {}
+    failed = {}
+    file = pd.ExcelFile(src)
+    sheets_to_load = list(sheets or file.sheet_names)
+    for sheet_name in sheets_to_load:
+        try:
+            df = pd.read_excel(file, sheet_name=sheet_name, **kwargs).squeeze("columns")
+        except Exception as e:
+            _logger.exception("Loading failed for sheet %r", sheet_name)
+            _logger.info("An empty dataframe will be used as fallback.")
+            failed[sheet_name] = e
+            df = pd.DataFrame()
+        finally:
+            out[sheet_name] = df
+    if failed:
+        exc = DataLoadingError(failed)
+        if raises:
+            raise exc
+        else:
+            warnings.warn(exc.summary)
```

Comment:
```suggestion
            warnings.warn(exc.summary + "\nFor these sheets, an empty dataframe is used as fallback.\n")
```
If execution will continue after the warning, I think it's good to include a message for the user explaining the fallback action.

Reply:
I like the suggestion, thanks!